### DISCUSSION TITLE
Detect if a menu has just been opened or not

### DISCUSSION THREAD

**Discussion by codecat** [ocornut/imgui#3952] [Old/Archived discussions]
If I begin a menu with `BeginMenu`, is there a way to check if it had **just** been opened or if it was already open on the previous frame? I'd like to populate the items of the menu dynamically, but don't want to do it every single rendered frame.

I can think of this as a way to do it, but it's kind of hacky:

```c++
static bool _hasItems = false;
if (ImGui::BeginMenu("Items")) {
  if (!_hasItems) {
    // populate items
    _hasItems = true;
  }
  ImGui::EndMenu();
} else {
  _hasItems = false;
}
```

Is there a more ImGui-native way to do this?

**Comment by ocornut** [comment]
You can use IsWindowAppearing()

**Reply by codecat** [reply]
Awesome, thank you.
